yb-voyager/src/query/queryissue/detectors.go

```diff
@@ -391,6 +393,46 @@ func (d *JsonQueryFunctionDetector) GetIssues() []QueryIssue {
 	return issues
 }
 
+type UniqueNullsNotDistinctDetector struct {
+	query    string
+	detected bool
+}
+
+func NewUniqueNullsNotDistinctDetector(query string) *UniqueNullsNotDistinctDetector {
+	return &UniqueNullsNotDistinctDetector{
+		query: query,
+	}
+}
+
+// Detect checks if a unique constraint is defined which has nulls not distinct
+func (d *UniqueNullsNotDistinctDetector) Detect(msg protoreflect.Message) error {
+	if queryparser.GetMsgFullName(msg) == queryparser.PG_QUERY_INDEX_STMT_NODE {
+		proto := msg.Interface().(proto.Message)
+		indexStmt := proto.(*pg_query.IndexStmt)
+
+		if indexStmt.Unique && indexStmt.NullsNotDistinct {
+			d.detected = true
+		}
+	} else if queryparser.GetMsgFullName(msg) == queryparser.PG_QUERY_TABLECONSTRAINT_NODE {
```

Comment:
Will this node come up for both ALTER TABLE ADD CONSTRAINT as well as CREATE TABLE ?

Reply:
yep it does

Reply:
nice. I was initially going to suggest we write this code inside `tableIssueDetector` and `alterTableIssueDetector`, but this is useful, since we just have to find the constraint node and check. 
cc: @priyanshi-yb